### PR TITLE
Di 448

### PR DESCRIPTION
**Pull request by mtagle**
also some style fixes that I had previously missed.

### CODE REVIEW

**Comment by mtagle** [comment]
so right now I have the retry logic for quota exceeded as separate from the retry logic for 500/503.  This is kind of confusing, but it seemed justified to me.  I'm open to opinions though. (Or Ideas on how I can unify them a bit)

also, technically 403 covers more than just quota exceeded errors.  Right now I'm just blindly assuming that any 403 should be retried and doing that. There are a couple of 403s that probably shouldn't be, but they seem like very edge casey cases: accessDenied, billingNotEnabled, blocked, and responseTooLarge (this shouldn't ever happen with what we're doing, as far as I know). There is also rateLimitExceeded, but I put that in the same bucket as quotaExceeded.

**Comment by mtagle** [comment]
used this SO question to aid in figuring out how to tell quotaExceeded 403 from others: http://stackoverflow.com/questions/38573446/bigquery-decyphering-quota-exceeded-message (because of course this doesn't seem to be documented by google...)

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java [hunk 1]
```diff
@@ -156,7 +154,6 @@ public Void call() throws InterruptedException {
       List<List<RowToInsert>> partitions = rowPartitioner.partition(rows);
       bigQueryWriter.writeRows(table, partitions.get(0), topic, schemas);
```
Comment (criccomini) [inline kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java]: Clean this loop

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/BigQueryWriter.java [hunk 2]
```diff
@@ -123,10 +133,24 @@ public void writeRows(
         if (err.code() == INTERNAL_SERVICE_ERROR || err.code() == SERVICE_UNAVAILABLE) {
           // backend error: https://cloud.google.com/bigquery/troubleshooting-errors
           retryCount++;
+        } else if (err.code() == FORBIDDEN
+                   && err.error() != null
+                   && QUOTA_EXCEEDED_REASON.equals(err.error().reason())) {
+          // quota exceeded error
```
Comment (criccomini) [inline kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/BigQueryWriter.java]: Would be good to have at least a debug log here, so we know when we're hitting quotas.

Also would be good to have a metric to count retries.

**Comment by criccomini** [comment]
LGTM just a couple of nits

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java (hunk 1 above)
Comment (mtagle) [inline kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java]: oops, ty

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/BigQueryWriter.java [hunk 3]
```diff
@@ -72,6 +87,21 @@ public BigQueryWriter(int retries, long retryWaitMs, Metrics metrics) {
...
+    requestRetries.add(metrics.metricName("request-retries-avg",
+                                          MetricsConstants.groupName,
+                                          "The average number of retries per request"),
+                       new Avg());
+    requestRetries.add(metrics.metricName("request-retries-max",
+                                          MetricsConstants.groupName,
+                                          "The maximum number of retry attempts made for a single "
+                                          + "request"),
+                       new Max());
+    requestRetries.add(metrics.metricName("request-retires-count",
```
Comment (criccomini) [inline kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/BigQueryWriter.java]: retires -> retries

**Pull request merged**